### DISCUSSION TITLE
HTTP plugin: `serve_http`: invalid multiple values to key `date` in tree

### DISCUSSION THREAD

**Discussion by FluffyDiscord** [roadrunner-server/roadrunner#1618] [Q&A]
### No duplicates 🥲.

- [X] I have searched for a similar issue in our bug tracker and didn't find any solutions.

### What happened?

HTTP server seems o fail to handle the following payload:
![image](https://github.com/roadrunner-server/roadrunner/assets/10332247/fb29b810-bb3a-4872-9417-be9039a342ee)

I does make sense as this is user error, but it should not return status code 500 with plaintext message `serve_http: invalid multiple values to key 'XXX' in tree`, but instead silently skip that invalid key. This error is not logged inside PHP since it wont even reach the PHP runtime.

If this is intended behavior, please add option we can toggle and silently skip this error. I would rather get half of the input that I can validate than nothing at all.

### Version (rr --version)

2.12.3

### How to reproduce the issue?

payload source, POST FormData
```
------WebKitFormBoundaryl6lIWCjXEqjruvDz
Content-Disposition: form-data; name="duplicate_daily_record_day[date][date]"

2023-06-16
------WebKitFormBoundaryl6lIWCjXEqjruvDz
Content-Disposition: form-data; name="duplicate_daily_record_day[sectionsToCopy][]"

PERMISSION_TYPE_DAILY_RECORD_ON_SITE_MECHANISM
------WebKitFormBoundaryl6lIWCjXEqjruvDz
Content-Disposition: form-data; name="duplicate_daily_record_day[date]"

2023-06-14 00:00
------WebKitFormBoundaryl6lIWCjXEqjruvDz
Content-Disposition: form-data; name="duplicate_daily_record_day[_token]"

86.ASFz-b72sEhJDViwjmrOEEiYg4FpsOpfGEEA8M-FaZ4.NUgXg8-65gR5WhX9wFi0UyTB9dEj5r9uSXdTwqDJRPB3SRK92L7GPgVdEQ
------WebKitFormBoundaryl6lIWCjXEqjruvDz--
```

**Comment by rustatian** [comment]
Hey @FluffyDiscord 👋🏻 
Could you please check the latest stable version? `v2023.1.5`. I remember that I fixed similar bug.

**Comment by rustatian** [comment]
> I does make sense as this is user error, but it should not return status code 500 with plaintext message serve_http: invalid multiple values to key 'XXX' in tree, but instead silently skip that invalid key.

You may override internal error code to [any code you need](https://github.com/roadrunner-server/roadrunner/blob/master/.rr.yaml#L616).

I'm against the practice of silently skipping errors. Since RR cannot parse the payload with errors, errors should be returned explicitly.

Another question is: How to handle duplicates? As far as I remember, according to the [RFC 7578](https://datatracker.ietf.org/doc/html/rfc7578) there is no clear answer on that.

**Reply by akeylimepie** [reply]
> You may override internal error code to [any code you need](https://github.com/roadrunner-server/roadrunner/blob/master/.rr.yaml#L616).

For this case, a 500 error is hardcoded:
https://github.com/roadrunner-server/http/blob/722fc5971e81881d7f9b714eedf5c5d777780b9b/handler/handler.go#L140

**Comment by rustatian** [comment]
I'm moving this issue to `Discussion` since this is not a bug, but rather a question that needs to be discussed.

P.S. Please attach a curl/postman request to your issue to clearly reproduce this behavior..
